### PR TITLE
CN Test Selection Page Student Create Quiz

### PR DESCRIPTION
**Pull request by chrisn02**
The student is now able to create their own quiz on the test selection page. It works by clicking the "Create Quiz" button, which will cause a modal to popup, which asks the user to select the subject and the number of questions, with the options of 10, 15 and 20 questions. It will select the questions that will be included in the quiz by random. Once the quiz has been created than a popup will appear saying "Quiz successfully created"

image
image
image
image

### CODE REVIEW

File: php/studentAssignQuiz.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    $conn->beginTransaction();
+
+    //Query to create the quiz
+    $stmt = $conn->prepare("CALL StudentCreateQuiz(:quizID, :subjectID, :title, :available)");
+    $quizID = bin2hex(random_bytes(16));
+    $stmt->bindValue(":quizID", $quizID, PDO::PARAM_STR);
+    $stmt->bindValue(":subjectID", $subjectID, PDO::PARAM_STR);
+    $stmt->bindValue(":title", $title, PDO::PARAM_STR);
+    $stmt->bindValue(":available", $available, PDO::PARAM_STR);
```
Comment (Jforjo) [inline php/studentAssignQuiz.php:30]: This should be null, and I believe PDO has a `PARAM_NULL` for that too. An available time of NULL just means that is always available.

**Review by Jforjo — APPROVED** [review]
Looks good, although I think the number of questions could simply be a number field rather than a select amount of options (up to you). You should also check whether there are enough questions for that particular subject. Wouldn't want a student to create a subject with 20 questions, unbeknownst to them that there is only 1 question for that subject, and is therefore met with 20 of the same questions.

**Comment by chrisn02** [comment]
Thanks. I will add the check to see if the subject has less questions than what the user chose, but I would imagine for the final product every subject will have at least 20 questions

**Review by gchodancar19 — APPROVED** [review]
This looks absolutely perfect as some of things which I have mentioned in previous comments have been mention like "select a subject" has got a box formatted design.

**Comment by chrisn02** [comment]
> This looks absolutely perfect as some of things which I have mentioned in previous comments have been mention like "select a subject" has got a box formatted design.

That's something I will check out, thanks

**Pull request merged**